internal/host/validations_test.go

```diff
@@ -2605,4 +2605,134 @@ var _ = Describe("Validations test", func() {
+					{MtuSuccessful: false, RemoteIPAddress: "192.168.145.31", OutgoingNic: "ens4"},
+				},
+			})
+
+			report, err := hostutil.MarshalConnectivityReport(connectivityReport)
+			Expect(err).To(BeNil())
+			host.Connectivity = report
+			Expect(db.Create(&cluster).Error).ToNot(HaveOccurred())
+			Expect(db.Create(&host).Error).ShouldNot(HaveOccurred())
+
+			mockAndRefreshStatus(&host)
+
+			refreshedHost := hostutil.GetHostFromDB(*host.ID, host.InfraEnvID, db).Host
+
+			status, _, ok := getValidationResult(refreshedHost.ValidationsInfo, IsMtuOKInMachineNetwork)
+			Expect(ok).To(BeTrue(), fmt.Sprintf("debuuging info: status %s, isok: %t host: %#v cluster: %#v", status, ok, host, cluster))
+			Expect(status).To(Equal(ValidationSuccess))
+		})
+		It("Bad flow - MTU reports in the machine network are unsuccessful", func() {
+			connectivityReport := &models.ConnectivityReport{}
+			connectivityReport.RemoteHosts = append(connectivityReport.RemoteHosts, &models.ConnectivityRemoteHost{
+				HostID: hostID,
+				MtuReport: []*models.MtuReport{
+					{MtuSuccessful: false, RemoteIPAddress: "192.168.127.31", OutgoingNic: "ens3"},
+				},
+			})
+
+			report, err := hostutil.MarshalConnectivityReport(connectivityReport)
+			Expect(err).To(BeNil())
+			host.Connectivity = report
+			Expect(db.Create(&cluster).Error).ToNot(HaveOccurred())
+			Expect(db.Create(&host).Error).ShouldNot(HaveOccurred())
+
+			mockAndRefreshStatus(&host)
+
+			refreshedHost := hostutil.GetHostFromDB(*host.ID, host.InfraEnvID, db).Host
+
+			status, _, ok := getValidationResult(refreshedHost.ValidationsInfo, IsMtuOKInMachineNetwork)
+			Expect(ok).To(BeTrue(), fmt.Sprintf("debuuging info: status %s, isOK: %t host: %#v cluster: %#v", status, ok, host, cluster))
+			Expect(status).To(Equal(ValidationFailure))
+		})
+	})
+	Context("Is MTU OK - UMN (Verify All Reports)", func() {
+		var (
+			cluster common.Cluster
+			host    models.Host
+		)
+		BeforeEach(func() {
+			cluster = hostutil.GenerateTestCluster(clusterID)
+			hostId, infraEnvId := strfmt.UUID(uuid.New().String()), strfmt.UUID(uuid.New().String())
+			mockProviderRegistry.EXPECT().IsHostSupported(commontesting.EqPlatformType(models.PlatformTypeVsphere), gomock.Any()).Return(false, nil).AnyTimes()
+			host = hostutil.GenerateTestHostByKind(hostId, infraEnvId, &clusterID, models.HostStatusKnown, models.HostKindHost, models.HostRoleMaster)
+			host.Inventory = hostutil.GenerateMasterInventory()
+			isUMN := true
+			cluster.UserManagedNetworking = &isUMN
+		})
+		It("Happy flow - all Mtu reports successful", func() {
+			connectivityReport := &models.ConnectivityReport{}
+			connectivityReport.RemoteHosts = append(connectivityReport.RemoteHosts, &models.ConnectivityRemoteHost{
+				HostID: hostID,
+				MtuReport: []*models.MtuReport{
+					{MtuSuccessful: true, RemoteIPAddress: "192.168.127.31", OutgoingNic: "ens3"},
+					{MtuSuccessful: true, RemoteIPAddress: "192.168.145.31", OutgoingNic: "ens4"},
```

Comment:
I would like to see also  IPv6 and Dual stack tests

Reply:
Just to clarify my understanding—why is this important here? I agree that checking IPv6 on the agent side is necessary, but in this case, we aren't concerned with whether it's IPv6 or IPv4. What matters here is whether the report  successful or not.